[Mono] Update about/warning text

aaronfranke opened the pull request:
I've removed the section about being unable to export games using C# - as you are now able to do this, as long as the export templates are installed. Also, I've made a minor grammar tweak.

YeldhamDev commented:
Could you also TTR it? That would close #17228.

akien-mga commented on modules/mono/editor/godotsharp_editor.cpp:
```diff
@@ -278,13 +278,11 @@ GodotSharpEditor::GodotSharpEditor(EditorNode *p_editor) {
 				String("C# support in Godot Engine is a brand new feature and a work in progress.\n") +
-				"It is at the alpha stage and thus not suitable for use in production.\n\n" +
+				"It is currently in an alpha stage and is not suitable for use in production.\n\n" +
 				"As of Godot 3.0, C# support is not feature-complete and may crash in some situations. " +
```
This and the next line also need to be updated.

aaronfranke replied:
Done. I only kept the "3.x" on the next line, as "3.1.x" is kinda implied by "3.x", and even if it isn't, the majority of C# changes will be introduced in the "3.x.0" releases.

aaronfranke committed:
[Mono] Update about/warning text

I've removed the section about being unable to export games using C# - as you are now able to do this, as long as the export templates are installed. Also, I've made a few minor grammar tweaks.

akien-mga commented:
Thanks!

akien-mga commented:
@hpvb Should be cherry-picked with `s/3.1/3.0.x/`.

hpvb commented:
Cherry picked into 3.0.6
